helix-term/src/commands.rs

```diff
@@ -4624,6 +4624,13 @@ fn keep_or_remove_selections_impl(cx: &mut Context, remove: bool) {
                 selection::keep_or_remove_matches(text, doc.selection(view.id), &regex, remove)
             {
                 doc.set_selection(view.id, selection);
+            } else {
+                if remove {
+                    cx.editor.set_status("no selection to remove");
+                } else {
+                    doc.reset_selection(view.id);
+                    cx.editor.set_status("no matches found");
+                }
```

Comment:
```suggestion
                cx.editor.set_error("no selections remaining");
```

A statusline message when there are no matches would be a nice improvement. Kakoune's behavior is to set an error message in the statusline saying "no selections remaining" (the equivalent here is `editor.set_error("no selections remaining")`).

We shouldn't be resetting the selections with `Document::reset_selection` though. That's only meant to setup a default selection for a document on a new View and should probably be a private function.

Reply:
Thanks!
Added your suggestion.